testdata/p4_14_samples_outputs/issue894-midend.p4

```diff
@@ -150,13 +150,13 @@ control ingress(inout headers hdr, inout metadata meta, inout standard_metadata_
         hash<bit<16>, bit<16>, tuple_0, bit<32>>(meta.custom_metadata.hash_val1, HashAlgorithm.csum16, 16w0, { hdr.ipv4.srcAddr, hdr.ipv4.dstAddr, hdr.ipv4.protocol, hdr.tcp.srcPort, hdr.tcp.dstPort }, 32w16);
         heavy_hitter_counter1.read(tmp_6, (bit<32>)meta.custom_metadata.hash_val1);
         meta.custom_metadata.count_val1 = tmp_6;
-        meta.custom_metadata.count_val1 = meta.custom_metadata.count_val1 + 16w1;
-        heavy_hitter_counter1.write((bit<32>)meta.custom_metadata.hash_val1, meta.custom_metadata.count_val1);
+        meta.custom_metadata.count_val1 = tmp_6 + 16w1;
+        heavy_hitter_counter1.write((bit<32>)meta.custom_metadata.hash_val1, tmp_6 + 16w1);
```

Comment:
Good question -- it depends whether adders or wires and dependencies are more expensive.  It could be converted back to a copy by CSE; any given target will need to balance which way is better.